File: Examples/VisualBasic/ModifyingAndConvertingImages/DICOM/AdjustGammaDICOM.vb

Imports Aspose.Imaging.FileFormats.Dicom
Imports Aspose.Imaging.FileFormats.Tiff
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.DICOM
    Class AdjustGammaDICOM
        Public Shared Sub Run()
            ' ExStart:AdjustGammaDICOM
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_DICOM()

            ' Load a DICOM image in an instance of DicomImage
            Using image As New DicomImage(dataDir & Convert.ToString("image.dcm"))
                ' Adjust the gamma
                image.AdjustGamma(50)

                ' Create an instance of BmpOptions for the resultant image and Save the resultant image
                image.Save(dataDir & Convert.ToString("AdjustGammaDICOM_out.bmp"), New BmpOptions())
            End Using
            ' ExEnd:AdjustGammaDICOM
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/DICOM/AdjustGammaDICOM.cs
// GIST-ID: 65ea7da6ed4ae23c4750a05d4a44eaef
using Aspose.Imaging.FileFormats.Dicom;
using Aspose.Imaging.ImageOptions;
using System;
using System.IO;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.DICOM
{
    class AdjustGammaDICOM
    {
        public static void Run()
        {
            // ExStart:AdjustGammaDICOM
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_DICOM();

            Console.WriteLine("Running example AdjustGammaDicom");
            using (var fileStream = new FileStream(dataDir + "file.dcm", FileMode.Open, FileAccess.Read))
            using (DicomImage image = new DicomImage(fileStream))
            {
                // Adjust the gamma, create an instance of BmpOptions for the resultant image, and save the resultant image.
                image.AdjustGamma(50);
                image.Save(dataDir + "AdjustGammaDICOM_out.bmp", new BmpOptions());
            }

            Console.WriteLine("Finished example AdjustGammaDicom");
            // ExEnd:AdjustGammaDICOM
        }
    }
}